### DISCUSSION TITLE
Confused about starting the browser

### DISCUSSION THREAD

**Discussion by stevebeauge** [aklinker1/vite-plugin-web-extension#109] [Q&A] [answered]
I'd like to start the Microsoft Edge Chromium when running `pnpm run dev`.

The page https://vite-plugin-web-extension.aklinker1.io/guide/configure-browser-startup.html is quite obscure. There's no details regarding the options, and the linked page which targets the source code of webext does not help much.

How should I setup a webextrc.json file to start ms edge?

FYI, I tried using `webextrc.json` like this:

````
{
    "target" : "chromium",
    "chromium-binary": "C:\\Program Files (x86)\\Microsoft\\Edge\\Application\\msedge.exe",
    "args" : "--profile-directory=\"Profile 2\""
}
````

But when I run `pnpm run dev` it fails with:

    message: 'No Chrome installations found.',
    code: 'ERR_LAUNCHER_NOT_INSTALLED'

Thanks

**Comment by stevebeauge** [comment]
A bit of progress:

1. The filename is `.webextrc.json` and not `webextrc.json`
2. the option `chromium-binary` is incorrect and should be `chromiumBinary`
3. `args` should be an array, not a string

The fixed file is:

````json
{
    "target": "chromium",
    "chromiumBinary": "C:\\Program Files (x86)\\Microsoft\\Edge\\Application\\msedge.exe",
    "args": [
        "--profile-directory=\"Profile 2\""
    ]
}
````

However, it's still not working. Edge is launching, but an error is throw (cannot load profile)

**Comment by aklinker1** [comment]
`web-ext` doesn't provide any docs for their JS API, and I'm not going to maintain a separate list. Right now, the [`webExtConfig` docs](https://vite-plugin-web-extension.aklinker1.io/config/plugin-options.html#webextconfig) has a link to the options, but you're right, I don't ellaborate on how to find them.

The best thing I've found is to look at the [CLI flags available for `web-ext run`](https://extensionworkshop.com/documentation/develop/web-ext-command-reference/#web-ext-run), and convert them from `kebab-case` to `camelCase`.

In your case:

1. You're not escaping the `C:\\` correctly, it should be `C:\\\\` to result in two forward slashes. Personally, I prefer YAML (`.webextrc.yml`) to avoid these mistakes
2. Target defaults to `chromium`, so you can remove that.
3. There's a separate CLI flag for selecting a profile. Use that instead of `args`.

```json
{
  "chromiumBinary": "C:\\\\Program Files (x86)\\Microsoft\\Edge\\Application\\msedge.exe",
  "chromiumProfile": "C:\\\\full\\path\\to\\Profile 2",
}
```

Or in YAML:

```yml
chromiumBinary: C:\\Program Files (x86)\Microsoft\Edge\Application\msedge.exe
chromiumProfile: C:\\full\path\to\Profile 2
```

I'll update the docs!

**Comment by aklinker1** [comment, marked as answer]
New docs: 
- PR: #110
- Preview: https://vite-plugin-web-extension-docs-1scmnez75-aklinker1.vercel.app/guide/configure-browser-startup.html

@stevebeauge Let me know if you have any feedback, or I'll merge the PR in the next few days.

**Reply by stevebeauge** [reply]
Thanks !

The update clarifies a lot how to construct the file.
